packages/back-end/src/services/experiments.ts

```diff
@@ -910,3 +921,131 @@ export function toMetricApiInterface(
+    return { regressionAdjustmentEnabled, metricRegressionAdjustmentStatuses };
+  }
+
+  const allExperimentMetricIds = uniq([
+    ...experiment.metrics,
+    ...(experiment.guardrails ?? []),
+    ...(experiment.activationMetric ?? []),
+  ]);
+  const allExperimentMetrics = await getMetricsByIds(
+    allExperimentMetricIds,
+    organization.id
+  );
+  const denominatorMetricIds = uniq(
+    allExperimentMetrics.map((m) => m.denominator).filter((m) => m)
+  ) as string[];
+  const denominatorMetrics = await getMetricsByIds(
+    denominatorMetricIds,
+    organization.id
+  );
+
+  for (const metric of allExperimentMetrics) {
+    if (!metric) continue;
+    const {
+      metricRegressionAdjustmentStatus,
+    } = getRegressionAdjustmentsForMetric({
+      metric: metric,
+      denominatorMetrics: denominatorMetrics,
+      organizationSettings: organization.settings,
+      metricOverrides: experiment.metricOverrides,
+    });
+    if (metricRegressionAdjustmentStatus.regressionAdjustmentEnabled) {
+      regressionAdjustmentEnabled = true;
+    }
+    metricRegressionAdjustmentStatuses.push(metricRegressionAdjustmentStatus);
+  }
+  if (!experiment.regressionAdjustmentEnabled) {
+    regressionAdjustmentEnabled = false;
+  }
+  return { regressionAdjustmentEnabled, metricRegressionAdjustmentStatuses };
+}
+
+export function getRegressionAdjustmentsForMetric({
+  metric,
+  denominatorMetrics,
+  organizationSettings,
+  metricOverrides,
+}: {
+  metric: MetricInterface;
+  denominatorMetrics: MetricInterface[];
+  organizationSettings?: Partial<OrganizationSettings>; // can be RA fields from a snapshot of org settings
+  metricOverrides?: MetricOverride[];
+}): {
+  newMetric: MetricInterface;
+  metricRegressionAdjustmentStatus: MetricRegressionAdjustmentStatus;
+} {
+  const newMetric = cloneDeep<MetricInterface>(metric);
+
+  // start with default RA settings
+  let regressionAdjustmentEnabled = false;
+  let regressionAdjustmentDays = 14;
+  let reason = "";
+
+  // get RA settings from organization
+  if (organizationSettings?.regressionAdjustmentEnabled) {
+    regressionAdjustmentEnabled = true;
+    regressionAdjustmentDays =
+      organizationSettings?.regressionAdjustmentDays ??
+      regressionAdjustmentDays;
+  }
+
+  // get RA settings from metric
+  if (metric?.regressionAdjustmentOverride) {
+    regressionAdjustmentEnabled = !!metric?.regressionAdjustmentEnabled;
+    regressionAdjustmentDays = metric?.regressionAdjustmentDays ?? 14;
+  }
+
+  // get RA settings from metric override
+  if (metricOverrides) {
+    const metricOverride = metricOverrides.find((mo) => mo.id === metric.id);
+    if (metricOverride?.regressionAdjustmentOverride) {
+      regressionAdjustmentEnabled = !!metricOverride?.regressionAdjustmentEnabled;
+      regressionAdjustmentDays =
+        metricOverride?.regressionAdjustmentDays ?? regressionAdjustmentDays;
+    }
+  }
+
+  // final gatekeeping
+  if (regressionAdjustmentEnabled) {
```

Comment:
I have some of this logic in the SqlIntegration.ts file itself in my SQL changes (see: 
https://github.com/growthbook/growthbook/pull/1031/files#diff-929850ea8fdeb86bac06cfd07f3d493e309a318aa3d1514c00299aa5d33d5630R775-R798)

Do we need to have it outside of SqlIntegration, as you do here, so that we can work with displays in the front-end? 

If I understand that as your goal, then your approach seems better, because our options are (1) your approach, which has this one logic in one place and passes the result to `SqlIntegration.ts` and to the front end, or (2) my approach, where we'd have to have this logic in the SQL generating file *and* in the front-end to display the tooltip, etc..